python/ray/ml/predictors/integrations/rl/rl_predictor.py

```diff
@@ -0,0 +1,117 @@
+    ):
+        self.trainer = trainer
+        self.preprocessor = preprocessor
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, env: Optional[EnvType] = None, **kwargs
+    ) -> "Predictor":
+        """Create RLPredictor from checkpoint.
+
+        This method requires that the checkpoint was created with the Ray AIR
+        RLTrainer.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from.
+            env: Optional environment to instantiate the trainer with. If not given,
+                it is parsed from the saved trainer configuration instead.
+
+        """
+        with checkpoint.as_directory() as checkpoint_path:
+            trainer_class_path = os.path.join(checkpoint_path, RL_TRAINER_CLASS_FILE)
+            config_path = os.path.join(checkpoint_path, RL_CONFIG_FILE)
+            preprocessor_path = os.path.join(checkpoint_path, PREPROCESSOR_KEY)
+
+            if not os.path.exists(trainer_class_path):
+                raise ValueError(
+                    f"RLPredictor only works with checkpoints created by "
+                    f"RLTrainer. The checkpoint you specified is missing the "
+                    f"`{RL_TRAINER_CLASS_FILE}` file."
+                )
+
+            if not os.path.exists(config_path):
+                raise ValueError(
+                    f"RLPredictor only works with checkpoints created by "
+                    f"RLTrainer. The checkpoint you specified is missing the "
+                    f"`{RL_CONFIG_FILE}` file."
+                )
+
+            with open(trainer_class_path, "rb") as fp:
+                trainer_cls = cpickle.load(fp)
+
+            with open(config_path, "rb") as fp:
+                config = cpickle.load(fp)
+
+            checkpoint_data_path = None
+            for file in os.listdir(checkpoint_path):
+                if file.startswith("checkpoint") and not file.endswith(
+                    ".tune_metadata"
+                ):
+                    checkpoint_data_path = os.path.join(checkpoint_path, file)
+
+            if not checkpoint_data_path:
+                raise ValueError(
+                    f"Could not find checkpoint data in RLlib checkpoint. "
+                    f"Found files: {list(os.listdir(checkpoint_path))}"
+                )
+
+            if os.path.exists(preprocessor_path):
+                with open(preprocessor_path, "rb") as fp:
+                    preprocessor = cpickle.load(fp)
+            else:
+                preprocessor = None
+
+            config.get("evaluation_config", {}).pop("in_evaluation", None)
+            trainer = trainer_cls(config=config, env=env)
+            trainer.restore(checkpoint_data_path)
+
+            return RLPredictor(trainer=trainer, preprocessor=preprocessor)
+
+    def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
+        if isinstance(data, pd.DataFrame):
```

Comment:
Maybe adda comment here that data - in other cases - may also be more complex than just an obs. E.g. internal-state information, prev-actions, etc..

In this case, we should use `policy.compute_actions_from_input_dict([input-dict])` instead below.

Reply:
This is a good point - is there an easy example that requires this? As we're currently only testing with PPO. I'd love to get this working with more complex policies from the start.

Reply:
Updated after syncing offline